src/app/code/Komoju/Payments/Plugin/WebhookEventProcessor.php

```diff
@@ -97,18 +97,17 @@ public function processEvent()
 
             $this->order->save();
         } elseif ($this->webhookEvent->eventType() == 'payment.cancelled') {
-            $this->order->setState(Order::STATE_CANCELED);
-            $this->order->setStatus(Order::STATE_CANCELED);
-            
+            $this->order->cancel();
```

Comment:
I'm glad you found a cleaner way to do this, have to set the state _and_ the status to the same thing definitely looks confusing (and I wrote the code! 😛 )